beacon-chain/rpc/beacon_chain_server_test.go

```diff
@@ -107,3 +109,135 @@ func TestBeaconChainServer_ListValidatorBalancesOutOfRange(t *testing.T) {
 		t.Errorf("Expected error %v, received %v", wanted, err)
 	}
 }
+
+func TestBeaconChainServer_GetValidatorsNoPagination(t *testing.T) {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+
+	count := 100
+	validators := make([]*ethpb.Validator, 0, count)
+	for i := 0; i < count; i++ {
+		if err := db.SaveValidatorIndex([]byte{byte(i)}, i); err != nil {
+			t.Fatal(err)
+		}
+		validators = append(validators, &ethpb.Validator{PublicKey: []byte{byte(i)}})
+	}
+
+	if err := db.SaveState(
+		context.Background(),
+		&pbp2p.BeaconState{Validators: validators}); err != nil {
+		t.Fatal(err)
+	}
+
+	bs := &BeaconChainServer{
+		beaconDB: db,
+	}
+
+	got, err := bs.GetValidators(context.Background(), &ethpb.GetValidatorsRequest{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(validators, got.Validators) {
+		t.Fatal("Incorrect respond of validators")
+	}
+}
+
+func TestBeaconChainServer_GetValidatorsPagination(t *testing.T) {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+
+	count := 100
+	balances := make([]uint64, count)
+	validators := make([]*ethpb.Validator, 0, count)
+	for i := 0; i < count; i++ {
+		if err := db.SaveValidatorIndex([]byte{byte(i)}, i); err != nil {
+			t.Fatal(err)
+		}
+		balances[i] = uint64(i)
+		validators = append(validators, &ethpb.Validator{PublicKey: []byte{byte(i)}})
+	}
+
+	if err := db.SaveState(
+		context.Background(),
+		&pbp2p.BeaconState{Validators: validators, Balances: balances}); err != nil {
+		t.Fatal(err)
+	}
+
+	bs := &BeaconChainServer{
+		beaconDB: db,
+	}
+
+	tests := []struct {
+		req *ethpb.GetValidatorsRequest
+		res *ethpb.Validators
+	}{
+		{req: &ethpb.GetValidatorsRequest{PageToken: strconv.Itoa(1), PageSize: 3},
+			res: &ethpb.Validators{
+				Validators: []*ethpb.Validator{
+					{PublicKey: []byte{1}},
+					{PublicKey: []byte{2}},
+					{PublicKey: []byte{3}}},
+				NextPageToken: strconv.Itoa(4),
+				TotalSize:     3}},
+		{req: &ethpb.GetValidatorsRequest{PageToken: strconv.Itoa(50), PageSize: 6},
+			res: &ethpb.Validators{
+				Validators: []*ethpb.Validator{
+					{PublicKey: []byte{50}},
+					{PublicKey: []byte{51}},
+					{PublicKey: []byte{52}},
+					{PublicKey: []byte{53}},
+					{PublicKey: []byte{54}},
+					{PublicKey: []byte{55}}},
+				NextPageToken: strconv.Itoa(56),
+				TotalSize:     6}},
+		{req: &ethpb.GetValidatorsRequest{PageToken: strconv.Itoa(98), PageSize: 999},
+			res: &ethpb.Validators{
+				Validators: []*ethpb.Validator{
+					{PublicKey: []byte{98}},
+					{PublicKey: []byte{99}}},
+				NextPageToken: strconv.Itoa(0),
```

Comment:
Please add test case where the optional fields are omitted

Reply:
This is tested in `TestBeaconChainServer_GetValidatorsNoPagination`:

https://github.com/prysmaticlabs/prysm/blob/da1678ecc02155ac9bd17ebbe3d8a84df1fb9d50/beacon-chain/rpc/beacon_chain_server_test.go#L113

Reply:
Added